portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -75,4 +90,40 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     // the response. Should consider whether or not this issue should be handled
     // on the server side or not.
   }
-}
+  private void refreshData(HttpServletRequest request,
+                           HttpServletResponse response) {
+
+    int queryLimit;
+
+    String requestLimit = request.getParameter("limit");
+
+    try {
+      // Convert the input to an integer.
+      queryLimit = Integer.parseInt(requestLimit);
+      System.out.println("parse successful");
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + requestLimit +
+                         " use default");
+      queryLimit = 20;
+    }
+
+    // Reset the data.
+    userData.clear();
+    System.out.println(queryLimit);
```

Comment:
Consider adding a little more context to your log lines, because someone else (or you in the future) debugging the program may not know the significance of a bare number. (Internally we use a logging library that also attaches info like line number and severity, but IMO stdout is fine here.)